src/SFA.DAS.EmployerFinance.AcceptanceTests/Steps/FinanceSteps.cs

```diff
@@ -22,6 +34,76 @@ public FinanceSteps(IObjectContainer objectContainer, ObjectContext objectContex
+            });
+        }
+
+        [Given("the account has transactions")]
+        public Task GivenTheAccountHasTransactions(Table table)
+        {
+            return _objectContainer.ScopeAsync(c =>
+            {
+                var account = _objectContext.Get<Account>();
+                var empRef = _objectContext.GetEmpRef();
+                var submissionId = 999000101;
+
+                var transactionLines = (
+                    from r in table.Rows
+                    let t = (TransactionItemType)Enum.Parse(typeof(TransactionItemType), r["TransactionType"])
+                    let d = DateTime.Parse(r["DateCreated"])
+                    select new TransactionLineEntity
+                    {
+                        AccountId = account.Id,
+                        TransactionType = t,
+                        DateCreated = d,
+                        TransactionDate = d,
+                        Amount = decimal.Parse(r["Amount"]),
+                        EmpRef = t == TransactionItemType.Declaration ? empRef : null,
+                        SubmissionId = t == TransactionItemType.Declaration ? submissionId++ : (int?)null,
+                        PeriodEnd = r.ContainsKey("PeriodEnd") ? r["PeriodEnd"] : null
+                    })
+                    .ToList();
+
+                var testTransactionRepository = c.Resolve<ITestTransactionRepository>();
+
+                return testTransactionRepository.CreateTransactionLines(transactionLines);
+            });
+        }
+
+        [When("the expire funds process runs on (.*) with a (.*) month expiry period")]
+        public async Task WhenTheExpireFundsProcessRuns(DateTime date, int expiryPeriod)
+        {
+            var configuration = _objectContainer.Resolve<EmployerFinanceConfiguration>();
+            var currentDateTime = _objectContainer.Resolve<Mock<ICurrentDateTime>>();
+            var messageSession = _objectContainer.Resolve<IMessageSession>();
+            var account = _objectContext.Get<Account>();
+
+            currentDateTime.Setup(c => c.Now).Returns(date);
+            configuration.FundsExpiryPeriod = expiryPeriod;
+
+            await messageSession.Send(new ExpireAccountFundsCommand { AccountId = account.Id });
+
+            await _objectContainer.ScopeAsync(async c =>
+            {
+                var transactionRepository = c.Resolve<ITransactionRepository>();
+                var timeout = Debugger.IsAttached ? 10 * 60 * 1000 : StepTimeout;
+                var cancellationTokenSource = new CancellationTokenSource(timeout);
+                var isComplete = await transactionRepository.WaitForAllTransactionLinesInDatabase(account, cancellationTokenSource.Token);
+
+                if (!isComplete)
+                {
+                    throw new Exception($"The transactions have not been completely loaded within the allowed time ({timeout} msecs). Either they are still loading or something has failed.");
```

Comment:
We could instead explicitly fail the test using `Assert.Fail()`.

Reply:
Done